tests/pyret/tests/test-parse-errors.arr

```diff
@@ -74,7 +74,36 @@ end
 bad-number = [list:
   ".0",
   "~.0",
-  "foo.0"
+  "foo.0",
+  #"2.",
```

Comment:
Those are already caught as a "program ended too soon" rather than as a bad-number parse error.  Devising a sufficient lookahead to trap them as bad numbers would involve a very complicated regexp, and it may not be worth the inefficiency given the errors are already caught via another route.